src/Glpi/Form/Migration/FormMigration.php

```diff
@@ -106,6 +115,50 @@ public function getTypesConvertMap(): array
         ];
     }
 
+    public const PUBLIC_ACCESS_TYPE = 0;
+    public const PRIVATE_ACCESS_TYPE = 1;
+    public const RESTRICTED_ACCESS_TYPE = 2;
+
+    /**
+     * Get the class strategy to use based on the access type
+     *
+     * @return array Mapping between access type constants and strategy classes
+     */
+    public function getStrategyForAccessTypes(): array
+    {
+        return [
+            self::PUBLIC_ACCESS_TYPE => DirectAccess::class,
+            self::PRIVATE_ACCESS_TYPE => DirectAccess::class,
+            self::RESTRICTED_ACCESS_TYPE => AllowList::class
+        ];
+    }
+
+    /**
+     * Create the appropriate strategy configuration based on form access rights
+     *
+     * @param array $form_access_rights The access rights data from the database
+     * @return JsonFieldInterface The configuration object for the access control strategy
+     * @throws LogicException When no strategy config is found for the given access type
+     */
+    public function getStrategyConfigForAccessTypes(array $form_access_rights): JsonFieldInterface
+    {
+        $clean_ids = fn($ids) => array_unique(array_filter($ids, fn($id) => is_int($id)));
+
+        if (in_array($form_access_rights['access_rights'], [self::PUBLIC_ACCESS_TYPE, self::PRIVATE_ACCESS_TYPE])) {
+            return new DirectAccessConfig(
+                allow_unauthenticated: $form_access_rights['access_rights'] === self::PUBLIC_ACCESS_TYPE
+            );
+        } elseif ($form_access_rights['access_rights'] === self::RESTRICTED_ACCESS_TYPE) {
+            return new AllowListConfig(
+                user_ids: $clean_ids(json_decode($form_access_rights['user_ids'], true) ?? []),
+                group_ids: $clean_ids(json_decode($form_access_rights['group_ids'], true) ?? []),
+                profile_ids: $clean_ids(json_decode($form_access_rights['profile_ids'], true) ?? [])
```

Comment:
I think it is preferable to throw an exception if the JSON is invalid rather than continuing the execution.

```suggestion
                user_ids: $clean_ids(json_decode($form_access_rights['user_ids'], associative: true, flags: JSON_THROW_ON_ERROR)),
                group_ids: $clean_ids(json_decode($form_access_rights['group_ids'], associative: true, flags: JSON_THROW_ON_ERROR)),
                profile_ids: $clean_ids(json_decode($form_access_rights['profile_ids'], associative: true, flags: JSON_THROW_ON_ERROR))
```